src/components/ui/forms/two-step/step2.js

```diff
@@ -0,0 +1,56 @@
+import { Button, ButtonNoLink } from '@/ui/buttons';
+import { Select, Textarea } from '@/ui/form-elements';
+
+/**
+ * @description - Form step 2
+ * @param {event} handleBack - the event that handles the back button
+ * @param {*} register - the React Hook Form register prop that applies the validation on the fields
+ * @param {string} errors - handles the error messages if any
+ * @returns
+ */
+export const Step2 = ({ handleBack, register, errors }) => {
+  return (
+    <div className='grid gap-4 md:grid-cols-2 mt-12'>
+      {/* department dropdown */}
+      <div>
+        <Select
+          label='Tell us about your department'
+          name='aboutYourDepartment'
+          isRequired
+          options={['Software Engineering', 'Sales & Marketing', 'Design']}
+          register={{ ...register('aboutYourDepartment') }}
+          errorMessage={errors.aboutYourDepartment?.message}
+        />
+      </div>
```

Comment:
same as above - it's fine to use department as an example of how to use this component, but it seems weird to hardcode these strings into the component code itself.

To me it would make more sense that these were somehow passed into TwoStepForm when that element is used in the actual JSX on the form page. That way, you could be putting many different TwoStepForms on the same website without needing to duplicate the components - it would just be a different invocation of the same component.

Maybe I'm misunderstanding something about NextJS components here, I'm on thin ice, but that's how it intuitively seems to me.

Reply:
Or another way to look at this:

If I import these components into my project, I shouldn't need to _rewrite_ the components to fit my project, I should just be able to _integrate them_ in a way that fits my project. I shouldn't need to touch that component code at all.